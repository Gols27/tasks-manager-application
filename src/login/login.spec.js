import React from 'react';
import { Login } from './';
import { shallow, mount } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';


it('renders without crashing', () => {
  jest.spyOn(React, 'useEffect').mockImplementation(f => f());
  const wrapper = shallow(<Login />);

  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('renders with changed input', () => {
  jest.spyOn(React, 'useEffect').mockImplementation(f => f());
  const wrapper = shallow(<Login userLoginRequest={jest.fn()} apiFailed={{apiError: true}}/>);

  wrapper.find('[name="password"]').simulate('change', {target: {value : 'Admin'}});
  wrapper.find('[name="userId"]').simulate('change', {target: {value : 'Admin'}});
  wrapper.find('[name="login"]').simulate('submit', {preventDefault: jest.fn()});
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('renders with changed input', () => {
  jest.spyOn(React, 'useEffect').mockImplementation(f => f());
  const wrapper = mount(<Login userLoginRequest={jest.fn()} users={{userAuth: false}} />);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

