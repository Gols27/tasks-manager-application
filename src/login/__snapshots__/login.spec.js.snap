// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with changed input 1`] = `
<div
  className="login-form"
>
  <form
    name="login"
    onSubmit={[Function]}
  >
     
    <label
      className="alert"
    >
       
      something went wrong. Pleas try after sometime!
       
    </label>
    <div>
      <input
        name="userId"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value="Admin"
      />
    </div>
    <div>
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value="Admin"
      />
    </div>
    <button
      className="button"
      disabled={false}
      size="lg"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`renders with changed input 2`] = `
<Login
  userLoginRequest={[MockFunction]}
  users={
    Object {
      "userAuth": false,
    }
  }
>
  <div
    className="login-form"
  >
    <form
      name="login"
      onSubmit={[Function]}
    >
       
      <div>
        <input
          name="userId"
          onChange={[Function]}
          placeholder="username"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
          value=""
        />
      </div>
      <button
        className="button"
        disabled={true}
        size="lg"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</Login>
`;

exports[`renders without crashing 1`] = `
<div
  className="login-form"
>
  <form
    name="login"
    onSubmit={[Function]}
  >
     
    <div>
      <input
        name="userId"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <button
      className="button"
      disabled={true}
      size="lg"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
